### PR TITLE
#2245 Allow Lua scripts to be run from the Conversations floater

### PR DESCRIPTION
**Pull request by maxim-productengine**

### CODE REVIEW

File: indra/newview/llfloaterimnearbychat.cpp [hunk 1]
```diff
@@ -606,6 +608,13 @@ void LLFloaterIMNearbyChat::sendChat( EChatType type )
                 if(!LLGestureMgr::instance().triggerAndReviseString(utf8text, &utf8_revised_text))
                 {
                     utf8_revised_text = utf8text;
+                    // check if the message is /filename.lua and execute the Lua script
+                    static const boost::regex is_lua_script("^/.*\\.luau?$");
```
Comment (nat-goodspeed) [inline indra/newview/llfloaterimnearbychat.cpp:612]: I recommend using `ll_regex_match()` from llregex.h: `boost::regex_match()` can throw. `ll_regex_match()` defends against that case.

Also, now that `runScriptFile()` can pass command-line arguments to the subject Lua script, I would match even if there's more text beyond the `.lua` extension.

**Review by nat-goodspeed — CHANGES_REQUESTED** [review]

File: indra/newview/llfloaterimnearbychat.cpp (hunk 1 above)
Comment (maxim-productengine) [reply indra/newview/llfloaterimnearbychat.cpp]: Thanks. You're right about passing arguments.

File: indra/newview/llfloaterimnearbychat.cpp [hunk 2]
```diff
@@ -606,6 +608,13 @@ void LLFloaterIMNearbyChat::sendChat( EChatType type )
                 if(!LLGestureMgr::instance().triggerAndReviseString(utf8text, &utf8_revised_text))
                 {
                     utf8_revised_text = utf8text;
+                    // check if the message is /filename.lua and execute the Lua script
+                    static const boost::regex is_lua_script("^/.*\\.luau?(?:\\s+\\S+)*$");
```
Comment (nat-goodspeed) [inline indra/newview/llfloaterimnearbychat.cpp:612]: Wouldn't it be equivalent (and more readable) just to end the pattern string at `"...luau?"`

**Review by nat-goodspeed — APPROVED** [review]

File: indra/newview/llfloaterimnearbychat.cpp (hunk 2 above)
Comment (maxim-productengine) [reply indra/newview/llfloaterimnearbychat.cpp:612]: Currently it matches strings like `/filename.lua` or `/filename.lua arg1 arg2`. Changing pattern to "^/.*\\.luau?" will allow matches like `/filename.luaqwerty123'.

Comment (maxim-productengine) [reply indra/newview/llfloaterimnearbychat.cpp:612]: Changing pattern to `^/.*\\.luau?` won't allow matches like  `/filename.lua arg1 arg2`.

Comment (nat-goodspeed) [reply indra/newview/llfloaterimnearbychat.cpp:612]: Even if you omit the final `$`? I thought that would match any string prefix.

You're right about trailing characters though. You could use `"^/.*\\.luau?\\b"` ?

**Pull request merged**